applications/IgaApplication/custom_processes/snake_sbm_process.h

```diff
@@ -0,0 +1,256 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Nicolo' Antonelli
+//                   Andrea Gorgi
+//
+
+#pragma once
+
+// Project includes
+#include "containers/model.h"
+#include "includes/model_part.h"
+#include "spatial_containers/bins_dynamic.h"
+#include "processes/process.h"
+
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/**
+ * @class SnakeSbmProcess
+ * @brief Process class for implementing Snake-based Surrogate Boundary Method (SBM).
+ * This class provides various functions to create and manipulate surrogate boundaries
+ * for Iga models in Kratos.
+ */
+class KRATOS_API(IGA_APPLICATION) SnakeSbmProcess
```

Comment:
So you finally removed the template argument. Fine if we don't really need it  :ok: